config/consensus.go

```diff
@@ -1272,21 +1272,48 @@ func initConsensusProtocols() {
 
 	v35.ApprovedUpgrades[protocol.ConsensusV36] = 140000
 
+	v37 := v36
+	v37.ApprovedUpgrades = map[protocol.ConsensusVersion]uint64{}
+
+	// TEAL resources sharing
+	v37.LogicSigVersion = 9
+	v37.EnablePrecheckECDSACurve = true
+
+	Consensus[protocol.ConsensusV37] = v37
+
+	// v36 can be upgraded to v37, with an update delay of 7 days ( see calculation above )
+	v36.ApprovedUpgrades[protocol.ConsensusV37] = 140000
+
+	v38 := v37
+	v38.ApprovedUpgrades = map[protocol.ConsensusVersion]uint64{}
+
+	// enables state proof recoverability
+	v38.StateProofUseTrackerVerification = true
+	v38.EnableCatchpointsWithSPContexts = true
+
+	// online circulation on-demand expiration
+	v38.ExcludeExpiredCirculation = true
+
+	// TEAL additional features
+	v38.AppForbidLowResources = true
```

Comment:
LogicSIgVersion 9 and forbid low resources should be in the same version. (I don't know how quiickly it flips through both)

Reply:
all features moved to v38